### ISSUE TITLE
Optimal for videos under 500MB; not recommended for large files.

### ISSUE THREAD

**Issue by JonnyBurger** [remotion-dev/remotion#3817]
[From message posted on Discord by **0x_master**](https://discord.com/channels/809501355504959528/817306238811111454/1234875967516835870)
----
I guess for big video file it doesn't worth it, but for light video like < 500MB can be good

**Comment by JonnyBurger** [comment]
Idea:

Allow wrapping an asset in  a new API `proxyS3()`

`<Img src={proxyS3(staticFile('bigmovie.mp4'))}>`

If we are not rendering or we are not on a S3 Serve URL, this does nothing at all.

Otherwise, we replace the URL with a call to `/s3-proxy?url=[original]` (similar to offthreadVideo: https://github.com/remotion-dev/remotion/blob/65245b5dc1d8a842cdc043c2347788cfe655290d/packages/core/src/video/offthread-video-source.ts#L16)

Then on the server, we implement a handler for the proxy (similar to https://github.com/remotion-dev/remotion/blob/65245b5dc1d8a842cdc043c2347788cfe655290d/packages/renderer/src/offthread-video-server.ts#L100) which calls the Lambda SDK instead `getS3Client()`

But we should not implement this in `@remotion/renderer`, we should do it in @remotion/lambda or make a new package, and then add it through a "injection" into @remotion/renderer.

We also don't want a full plugin system yet, so I would say we add this to the internal options (`RenderInternals.prepareServer`)

**Comment by vikramhegde58** [comment]
@JonnyBurger Is there a way to avoid storing progress.json file on to s3 and use cloudflare r2 instead?

**Comment by JonnyBurger** [comment]
@vikramhegde58 not at the moment!

**Comment by gmmarc** [comment]
I was able to avoid all data transfer costs from S3 by setting up an S3 VPC Endpoint.

TL;DR the Remotion Lambda accesses the S3 URL via the AWS’s internal network and is treated as “in-region” data access, which has **no cost if the bucket and the Lambda are the same region.**

NOTE: you still pay per requests, but I believe it's something like $0.01 per 1000 GET requests.

Setup summary:

1. Attach the VPC to the Remotion lambda:
```shell
npx remotion lambda functions deploy \
  --vpc-subnet-ids subnet-xxx,subnet-yyy \
  --vpc-security-group-ids sg-zzz
```

NOTE: if the Lambda has already been deployed, this will NOT update the settings. You can either delete the Lambda and deploy again, or modify the existing Lambda from the AWS console or through CLI

2. Create an S3 VPC Endpoint in the same VPC as the Lambda. Use `Gateway` type and the same private route tables as the subnets you chose 

3. Update the policy of the S3 bucket if needed, so the S3 VPC Endpoint can access. Example:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": "*",
            "Action": [
                "s3:GetObject"
            ],
            "Resource": "arn:aws:s3:::your-assets-bucket-name/*",
            "Condition": {
                "StringEquals": {
                    "aws:SourceVpce": "vpce-xxxx"
                }
            }
        }
    ]
}
```

**Comment by JonnyBurger** [comment]
Basically a duplicate of #3839, it will be the same idea!
